web/ASC.Web.Common/src/components/MediaViewer/MediaViewer.js

```diff
@@ -0,0 +1,318 @@
+                    }
+                );
+            }
+        }
+
+    }
+    mapSupplied = {
+        ".aac": { supply: "m4a", type: audio },
+        ".flac": { supply: "mp3", type: audio },
+        ".m4a": { supply: "m4a", type: audio },
+        ".mp3": { supply: "mp3", type: audio },
+        ".oga": { supply: "oga", type: audio },
+        ".ogg": { supply: "oga", type: audio },
+        ".wav": { supply: "wav", type: audio },
+
+        ".f4v": { supply: "m4v", type: video },
+        ".m4v": { supply: "m4v", type: video },
+        ".mov": { supply: "m4v", type: video },
+        ".mp4": { supply: "m4v", type: video },
+        ".ogv": { supply: "ogv", type: video },
+        ".webm": { supply: "webmv", type: video },
+        ".wmv": { supply: "m4v", type: video, convertable: true },
+        ".avi": { supply: "m4v", type: video, convertable: true },
+        ".mpeg": { supply: "m4v", type: video, convertable: true },
+        ".mpg": { supply: "m4v", type: video, convertable: true }
+    };
+
+    canImageView = function (ext) {
+        return this.props.extsImagePreviewed.indexOf(ext) != -1;
+    };
+    canPlay = (fileTitle, allowConvert) => {
+
+        var ext = fileTitle[0] === "." ? fileTitle : this.getFileExtension(fileTitle);
+
+        var supply = this.mapSupplied[ext];
+
+        var canConv = allowConvert || this.props.allowConvert;
+
+        return !!supply && this.props.extsMediaPreviewed.indexOf(ext) != -1
+            && (!supply.convertable || canConv);
+    };
+
+    getFileExtension = (fileTitle) => {
+        if (typeof fileTitle == "undefined" || fileTitle == null) {
+            return "";
+        }
+        fileTitle = fileTitle.trim();
+        var posExt = fileTitle.lastIndexOf(".");
+        return 0 <= posExt ? fileTitle.substring(posExt).trim().toLowerCase() : "";
+    };
+
+    prevMedia = () => {
+
+        let currentPlaylistPos = this.state.playlistPos;
+        currentPlaylistPos--;
+        if (currentPlaylistPos < 0)
+            currentPlaylistPos = this.state.playlist.length - 1;
+
+        this.setState({
+            playlistPos: currentPlaylistPos
+        });
+
+    };
+
+    nextMedia = () => {
+
+        let currentPlaylistPos = this.state.playlistPos;
+        currentPlaylistPos = (currentPlaylistPos + 1) % this.state.playlist.length;
+
+        this.setState({
+            playlistPos: currentPlaylistPos
+        });
+    };
+    getOffset = () => {
+        if (this.detailsContainer.current && this.viewerToolbox.current) {
+            return this.detailsContainer.current.offsetHeight + this.viewerToolbox.current.offsetHeight;
+        } else {
+            return 0;
+        }
+    }
+    render() {
+
+        let currentPlaylistPos = this.state.playlistPos;
+        let currentFileId = this.state.playlist.length > 0 ? this.state.playlist.find(file => file.id === currentPlaylistPos).fileId : 0;
+
+        let fileTitle = this.state.playlist[currentPlaylistPos].title;
```

Comment:
Move this.state.playlist[currentPlaylistPos] to single const and use it to get title and src